jkube-kit/jkube-kit-vertx/src/test/java/org/eclipse/jkube/vertx/generator/VertxGeneratorTest.java

```diff
@@ -101,20 +90,23 @@ public void testWithMetrics() {
 
     @Test
     public void testWithInfinispanClusterManager() {
-      // Given
-      // @formatter:off
-      new Expectations() {{
-        project.getOutputDirectory(); result = new File("target/tmp/target").getAbsolutePath();
-        project.getDependencies(); result = Arrays.asList(infinispan, core);
-      }};
-      // @formatter:on
-      // When
-      Map<String, String> env = new VertxGenerator(context).getEnv(true);
-      // Then
-      assertThat(env)
-              .containsEntry("JAVA_OPTIONS", "-Dvertx.cacheDirBase=/tmp/vertx-cache -Dvertx.disableDnsResolver=true " +
-                      // Force IPv4
-                      "-Djava.net.preferIPv4Stack=true")
-              .containsEntry("JAVA_ARGS", "-cluster");
+      try (MockedStatic<JKubeProjectUtil> mockedJKubeProjectUtil = mockStatic(JKubeProjectUtil.class)) {
+        // Given
+        mockedJKubeProjectUtil
+            .when(() -> JKubeProjectUtil.hasResource(project, "META-INF/services/io.vertx.core.spi.cluster.ClusterManager"))
+            .thenReturn(true);
+        mockedJKubeProjectUtil
+            .when(() -> JKubeProjectUtil.hasDependency(project, "io.vertx", "vertx-infinispan"))
+            .thenReturn(true);
```

Comment:
Considering this: https://github.com/eclipse/jkube/pull/1508#issuecomment-1134224352

Then this line should be changed to what was before, to make all test cases consistent:
```java
when(project.getDependencies()).thenReturn(Arrays.asList(infinispan, core));
```

Reply:
I tried reverting it but I started getting this error:

```
java.lang.AssertionError: 
Expecting map:
 <{"JAVA_ARGS"="-cluster", "JAVA_OPTIONS"="-Dvertx.cacheDirBase=/tmp/vertx-cache -Dvertx.disableDnsResolver=true"}>
to contain:
 <[MapEntry[key="JAVA_OPTIONS", value="-Dvertx.cacheDirBase=/tmp/vertx-cache -Dvertx.disableDnsResolver=true -Djava.net.preferIPv4Stack=true"]]>
but could not find the following map entries:
 <[MapEntry[key="JAVA_OPTIONS", value="-Dvertx.cacheDirBase=/tmp/vertx-cache -Dvertx.disableDnsResolver=true -Djava.net.preferIPv4Stack=true"]]>
```
It looks like control is not going in this if block as expected by test:
https://github.com/eclipse/jkube/blob/7e849a7dd04466ea10684e0c493c3e5dcd5db1ea/jkube-kit/jkube-kit-vertx/src/main/java/org/eclipse/jkube/vertx/generator/VertxGenerator.java#L74


I think it's because now we're mocking static methods in JKubeProjectUtil, real `hasDependency`is not invoked.

Reply:
OK, we can keep it this way.
We could tune the mock to call the real method, but then the test preparation would be even more verbose.